src/Link.cpp

```diff
@@ -0,0 +1,14 @@
+#include "Link.hpp"
+
+#include <utility>
+#include <vector>
+#include <memory>
+
+Link::Link(std::shared_ptr<NetworkDevice> source, std::shared_ptr<NetworkDevice> destination, std::uint64_t distance_ns):
```

Comment:
See my suggestion above.

```suggestion
Link::Link(std::shared_ptr<NetworkDevice> src, std::shared_ptr<NetworkDevice> dest, std::uint64_t latency_ns):
```